spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerResultAdapter.java

```diff
@@ -26,7 +26,7 @@
  * annotated method is operating in declarative mode.
  * @author Marius Bogoevici
  */
-public interface StreamListenerResultAdapter<R, B> {
+public interface StreamListenerResultAdapter<R, B, D> {
```

Comment:
Sounds reasonable, but I don't know the whole context.

Not sure though why you are asking me since I'm not so good in this project at all...